lib/tapioca/commands/abstract_gem.rb

```diff
@@ -252,6 +262,26 @@ def expected_rbis
           .to_h { |gem| [gem.name, gem.version.to_s] }
       end
 
+      sig do
+        params(
+          gem_name: String,
+          resolved_dependencies: T::Array[Tapioca::Gemfile::GemSpec],
+        ).returns(T::Array[Tapioca::Gemfile::GemSpec])
+      end
+      def gem_with_dependencies(gem_name, resolved_dependencies = [])
+        gem = @bundle.gem(gem_name)
+        return [] if gem.nil?
+
+        gem_dependencies = gem.dependencies.map { |gem_spec| @bundle.gem(gem_spec.name) }.compact
```

Comment:
This method was tough to understand at first. Maybe renaming variables could help? `gem_dependencies` could be `direct_dependencies` and `resolved_dependencies` could be `gem_names`?